controllers/topics/index.js

```diff
@@ -155,21 +156,49 @@ const followTopicV2 = async (req, res) => {
     const {name} = req.body;
     const {token, userId} = req;
 
+    // logic get user sign and anonymous
+    let secondDetailUser;
+    let secondDetailUserId;
+    const detailTokenUser = await UsersFunction.findUserById(User, userId);
+    if (!detailTokenUser.is_anonymous) {
+      secondDetailUser = await UsersFunction.findAnonymousUserId(User, userId);
+    } else {
+      secondDetailUserId = await UsersFunction.findSignedUserId(User, userId);
+      secondDetailUser = await UsersFunction.findUserById(User, secondDetailUserId);
+    }
+
     TopicValidator.validatePutTopicFollow({name});
 
+    //Logic get topic
     const topicService = new TopicService(Topics);
     const topic = await topicService.getTopicByName(name);
     const {topic_id} = topic;
     const userTopicService = new UserTopicService(UserTopic, UserTopicHistory);
-    const result = await userTopicService.followTopic(userId, topic_id);
 
-    const message = await _afterPutTopic(result, token, userId, name);
+    //get follow status
+    const [getTokenUserStatus, getSecondUserStatus] = await Promise.all([
+      userTopicService.getFollowTopicStatus(detailTokenUser.user_id, topic_id),
+      userTopicService.getFollowTopicStatus(secondDetailUser.user_id, topic_id)
+    ]);
+
+    let data = [];
+    if (getTokenUserStatus) {
+      await userTopicService.unfollowTopic(detailTokenUser.user_id, topic_id);
+      data.push(await _afterPutTopic(true, token, detailTokenUser.user_id, name));
+    } else {
+      await userTopicService.followTopic(detailTokenUser.user_id, topic_id);
+      data.push(await _afterPutTopic(false, token, detailTokenUser.user_id, name));
+
+      if (getSecondUserStatus) {
+        await userTopicService.unfollowTopic(secondDetailUser.user_id, topic_id);
+        data.push(await _afterPutTopic(true, token, secondDetailUser.user_id, name));
+      }
+    }
 
     return res.status(200).json({
       status: 'success',
       code: 200,
-      data: !result,
-      message
+      data: data
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential security issue here. The code retrieves the user details based on whether the user is anonymous or signed in, and then performs follow/unfollow actions for both the token user and a second user. This could potentially allow a malicious user to manipulate the follow status of another user. It's important to ensure that users can only modify their own data unless they have explicit permission to do so.

```diff
-    let secondDetailUser;
-    let secondDetailUserId;
-    const detailTokenUser = await UsersFunction.findUserById(User, userId);
-    if (!detailTokenUser.is_anonymous) {
-      secondDetailUser = await UsersFunction.findAnonymousUserId(User, userId);
-    } else {
-      secondDetailUserId = await UsersFunction.findSignedUserId(User, userId);
-      secondDetailUser = await UsersFunction.findUserById(User, secondDetailUserId);
-    }
```
The above lines should be removed as they are fetching and manipulating data for a second user which is not appropriate.



<!-- This is an auto-generated comment by OSS CodeRabbit -->